src/compositor/mod.rs

```diff
@@ -69,12 +56,16 @@ impl<'a> Compositor<'a> {
     /// Display the supplied texture for the next frame.
     ///
     /// If `bounds` is None, the entire texture will be used. Lens distortion is handled by the OpenVR implementation.
-    pub fn submit(&self, eye: Eye, texture: &Texture, bounds: Option<&texture::Bounds>) -> Result<(), CompositorError> {
+    ///
+    /// # Safety
+    ///
+    /// The handles you supply must be valid and comply with the graphics API's synchronization requirements.
+    pub unsafe fn submit(&self, eye: Eye, texture: &Texture, bounds: Option<&texture::Bounds>) -> Result<(), CompositorError> {
```

Comment:
I'd really prefer to have safe public methods, can't we have the unsafe handling inside the function?